wandb/sdk/wandb_run.py

```diff
@@ -62,10 +62,15 @@
         BinaryIO,
         Tuple,
         Union,
-        NoReturn,
         Type,
         Callable,
     )
+
+    try:
+        from typing import NoReturn
+    except ImportError:
+        NoReturn = None  # type: ignore
```

Comment:
Just change the NoReturn later on to "NoReturn"

Reply:
Do you mean like:
```python
def exit(self, code: object = 0) -> "NoReturn":
     #####
```

That fails with `error: Name 'NoReturn' is not defined`

Reply:
That is correct,  with python3.0/3.1 that code should never be executed as it is converted to:
wandb/sdk_py27/wandb_run.py
which strips the annotation.

Reply:
Yes, but that still cause flake8 and mypy checks to fail with `error: Name 'NoReturn' is not defined`.

Reply:
Nvm, I'll try ignoring those

Reply:
We only should be running flake8/mypy on python3.6+
how are you getting errors?

Reply:
Maybe we are missing a specifier in tox.ini saying that py3.6 is the requirement, i think it just says python3 and it normally picks up py3.6+
noone should be using python 3.6.0/1. it is often restricted from setup.py.

Reply:
![image](https://user-images.githubusercontent.com/11006006/107257927-a13f2800-6a54-11eb-8335-0a284b2a6877.png)

Reply:
heh, thanks, i guess there are 3.6.1 users.. they arent using modern pytorch or tensorflow as that support was dropped year+ ago?

ok, lets keep 3.6.1 support, but the string annotation should just work.  our library runtime will work fine on 3.6.1 but our flake8 and mypy checking should only be run for 3.6.2+
See: https://pypi.org/project/torch/

Reply:
The string annotation will only work if there is actually a variable with the same name, else flake8 and mypy complains (`tox -e flake8`, `tox -e mypy`). I'm not running the tests on 3.6.1.

Checkout this diff:
https://github.com/wandb/client/commit/dcbeda7267fa5ec40132d173a152538e37b82ace

And the CI output:
https://app.circleci.com/pipelines/github/wandb/client/3256/workflows/91f0385d-513c-45bf-8d24-aaaee0c4fd79/jobs/37119

Reply:
Here is the PR I was thinking would be better:
https://github.com/wandb/client/pull/1848

Reply:
This makes NoReturn only used during typechecking (mypy).  runtime 3.6.0, 3.6.1 should be fine.